collect/cache/cuckooSentCache.go

```diff
@@ -164,13 +166,26 @@ func NewCuckooSentCache(cfg config.SampleCacheConfig, met metrics.Metrics) (Trac
 		return nil, err
 	}
 	dropped := NewCuckooTraceChecker(cfg.DroppedSize, met)
+	// we want to keep track of the most recent dropped traces so we can avoid
+	// checking them in the dropped filter, which can have contention issues
+	// under high load. So we use a cache with TTL to keep track of the most
+	// recent dropped trace IDs, which lets us avoid checking the dropped filter
+	// for them for a short period of time. This means that when a whole batch
+	// of spans from the same trace arrives late, we don't have to check the
+	// dropped filter for each one. Benchmarks indicate that the Set cache is
+	// maybe 2-4x faster than the cuckoo filter and it also avoids lock
+	// contention issues in the cuckoo filter, so in practical use saves more
+	// than that. The TTL in this cache is short, because it's refreshed on each
+	// request.
+	recentDroppedIDs := generics.NewSetWithTTL[string](3 * time.Second)
```

Comment:
Should we have a metric to check how big this cache can get?

Reply:
I thought about it, but I'm worried that it would make this too slow, and the point of this PR is to squeeze out a few more microseconds. 

The size can only get as big as "the number of dropped traces (not spans) in 3 seconds", which even in heavily loaded systems is a pretty safe number.